collector/lib/RateLimit.cpp

```diff
@@ -74,4 +74,19 @@ bool RateLimitCache::Allow(std::string key) {
   return limiter_->Allow(&pair.first->second);
 }
 
+CountLimiter::CountLimiter()
+    : count_(4096) {}
+
+CountLimiter::CountLimiter(int64_t count)
+    : count_(count) {}
+
+bool CountLimiter::Allow(std::string key) {
+  auto pair = cache_.emplace(std::make_pair(key, TokenBucket()));
```

Comment:
```suggestion
  auto pair = cache_.try_emplace(key);
```
This might be slightly more efficient as it only creates the `TokenBucket` if it needs to. `try_emplace` was introduced in C++ 17 so the other similar functions use `emplace` instead of `try_emplace`.